File: Assets/CustomScripts/Platform/movingPlatformBehavior.js
#pragma strict

var movingPlatform : MovingPlatform;
public var wayPoints:List.<GameObject>;
private var player : Transform;
public var speed : int = 3;
public var rotationSpeed : int = 45;
private var controller : CharacterController;

function Start () {
	this.movingPlatform = MovingPlatform(transform, speed, rotationSpeed, wayPoints);
}

function Update () {
	this.movingPlatform.move();
	this.movingPlatform.rotate();
	if (movingPlatform.playerContact){
		this.player.position += this.movingPlatform.getDeltaPos();
		this.movingPlatform.rotateOnPlatform(player, controller);
	}
}

function transferSpeed (playerTransform : Transform) : void {
	this.player = playerTransform;
	this.movingPlatform.playerContact = true;
	this.controller = this.player.GetComponent(CharacterController);
}
	
function noContact() : void {
	this.movingPlatform.playerContact = false;
}

File: Assets/CustomScripts/MovingPlatform/movingPlatformBehavior.cs
using System;
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class movingPlatformBehavior : MonoBehaviour {
	public MovingPlatform movingPlatform;
	public List<GameObject> wayPoints;
	public float speed = 3;
	public float rotationSpeed = 45;
	public bool rotateAround = false;
	public bool maintainUp = false;
	public bool enable = false;

	void Awake () {
		this.movingPlatform = new MovingPlatform(this.transform, speed, rotationSpeed, wayPoints);
		this.movingPlatform.maintainUp = maintainUp;
		enabled = enable;
	}
	
	void FixedUpdate () {
			this.movingPlatform.move();
			if (rotateAround)
				this.movingPlatform.rotateAround();
			else
				this.movingPlatform.rotate();
	}
}